scripts/setup-environment.sh

```
@@ -0,0 +1,108 @@
+PROJECT_ROOT="$(cd "$(dirname "$0")/.." && pwd)"
+cd "$PROJECT_ROOT"
+
+# Parse CLI -------------------------------------------------------------------
+NON_INTERACTIVE=false
+CUSTOM_PROVIDER=""
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --non-interactive|-y) NON_INTERACTIVE=true ; shift ;;
+    --provider)
+      CUSTOM_PROVIDER="$2" ; shift 2 ;;
```

Shellcheck (0.10.0)

[warning] CUSTOM_PROVIDER appears unused. Verify use (or export if used externally).

(SC2034)